uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Upgrader/Step/1022.php

```diff
@@ -9,13 +9,15 @@
    */
   public function upgrade_1022()
   {
-    civicrm_api3('ContactWorkPattern', 'get', [
-      'return' => ['id'],
-      'api.ContactWorkPattern.update' => [
-        'change_reason' => 1,
-        'id' => "\$value.id"
-      ],
-    ]);
+    $query = 'UPDATE civicrm_hrleaveandabsences_contact_work_pattern set change_reason =
+      (
+        SELECT cov.value FROM civicrm_option_group cog LEFT JOIN civicrm_option_value cov
```

Comment:
It doesn't make much difference to the query result, but it would be clearer to exchange the `FROM` table with the join table. You're selecting FROM `civicrm_option_value` and joining to `civicrm_option_group` to filter the result.